docs/current_docs/features/shell.mdx

```diff
@@ -0,0 +1,96 @@
+
+Here's an example of Dagger Shell in action:
+
+![Dagger Shell](/img/current_docs/features/shell-curl.gif)
+
+Here is another, more complex example:
+
+![Dagger Shell](/img/current_docs/features/shell-build.gif)
+
+:::note
+Dagger Shell allows you to execute Dagger Functions as native shell commands, using a Bash-compatible syntax. However, instead of calling a subprocess for execution, Dagger Shell directly interprets commands within the Dagger CLI and executes them in containers. This means that all your commands execute as API calls in a secure, sandboxed environment, rather than as processes with direct access to the host system.
+:::
+
+## Modes of execution
+
+Dagger Shell supports multiple ways to run commands:
+
+<Tabs groupId="shell">
+<TabItem value="Inline execution">
+```shell
+dagger -c 'container | from alpine | terminal'
+```
+</TabItem>
+<TabItem value="Standard input">
+```shell title="Type 'dagger' to start Dagger Shell in interactive mode and enter the command below."
+echo 'container | from alpine | terminal' | dagger
+```
+</TabItem>
+<TabItem value="Script">
+```shell
+dagger script.sh
+```
+</TabItem>
+<TabItem value="Interactive REPL">
+```shell title="Type 'dagger' to start Dagger Shell in interactive mode and enter the command below."
+container | from alpine | terminal
+```
+</TabItem>
+</Tabs>
+
+## Built-in help
+
+Dagger Shell provides extensive built-in help:
+
+```shell
+.help             # Shows available commands
+.help MODULE      # Shows description, arguments, and return type for module constructor
+.help FUNCTION    # Shows description, arguments, and return type for function
+MODULE | .help    # Shows available functions for constructor returned by MODULE
+FUNCTION | .help  # Shows available functions for object returned by FUNCTION
+```
+
+![Dagger Shell help](/img/current_docs/features/shell-help.gif)
+
+## Navigation
+
+Dagger Shell presents a virtual filesystem to the user. When Dagger Shell starts, it sets the working directory to an initial context, which defaults to the enclosing Git repository or Dagger module. The user can navigate the virtual filesystem by changing the working directory.
+
+The initial context can be a local filesystem path or a remote filesystem. For example, `dagger -m github.com/dagger/dagger/docs@v0.13.0` changes the context to `github.com/dagger/dagger@v0.13.0` and the working directory to `/docs`.
+
+:::note
+To safeguard access to the host filesystem, the context root cannot be escaped.
+:::
+
+## Modules
+
+If the working directory contains a Dagger module, that module is automatically loaded, and its Dagger Functions can be directly inspected (via `.help`) or invoked.
```

Comment:
It's worth mentioning that if only the core API is needed, the `-n` (`--no-mod`) flag can be provided which makes for a faster load because the CLI doesn't try to find and load the current module. This replaces `dagger core`.

If `-n` nor `-m` are provided and the current workdir is not in a module, it will fallback to loading the core API only. So it's like a `dagger call` that fallsback to `dagger core`.